hooks/codec_settings.py

```diff
@@ -27,6 +25,11 @@ def get_quicktime_settings(self, **kwargs):
         Returns a dictionary of settings to be used for the Write Node that generates
         the Quicktime in Nuke.
         """
+        # This hook file gets loaded even if overriden in sgtk_config.
+        # import causes error if not launched from nuke.
+        # Therefore, moving the import into the function.
+        import nuke
+
```

Comment:
Though the hook is overridden in sgtk_config, the module gets loaded and gives an import error if not run from nuke.
Traceback from maya:
```
    // Error: Shotgun loader: Cannot load plugin file '/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-reviewsubmission/v0.3.x/hooks/codec_settings.py'
    Traceback (most recent call last):
      File "/dd/tools/cent6_64/package/sgtk_core/2.13.1/python/tank/util/loader.py", line 55, in load_plugin
        module = imp.load_source(module_uid, plugin_file)
      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-reviewsubmission/v0.3.x/hooks/codec_settings.py", line 18, in <module>
        import nuke
    ImportError: No module named nuke // 
    // Error: Shotgun dialog: Publish error stack:
    Traceback (most recent call last):
      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-publish2/v2.0.x/python/tk_multi_publish2/dialog.py", line 1041, in do_publish
        self._visit_tree_r(parent, lambda child: child.publish(), "Publishing")
      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-publish2/v2.0.x/python/tk_multi_publish2/dialog.py", line 1163, in _visit_tree_r
        action_name
      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-publish2/v2.0.x/python/tk_multi_publish2/dialog.py", line 1163, in _visit_tree_r
        action_name
      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-publish2/v2.0.x/python/tk_multi_publish2/dialog.py", line 1163, in _visit_tree_r
        action_name
      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-publish2/v2.0.x/python/tk_multi_publish2/dialog.py", line 1152, in _visit_tree_r
        status = action(child)  # eg. child.validate(), child.publish() etc.
      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-publish2/v2.0.x/python/tk_multi_publish2/dialog.py", line 1041, in <lambda>
        self._visit_tree_r(parent, lambda child: child.publish(), "Publishing")
      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-publish2/v2.0.x/python/tk_multi_publish2/publish_tree_widget/tree_node_task.py", line 148, in publish
        self._task.publish()
      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-publish2/v2.0.x/python/tk_multi_publish2/processing/task.py", line 177, in publish
        self.plugin.run_publish(self.settings, self.item)
      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-publish2/v2.0.x/python/tk_multi_publish2/processing/plugin.py", line 550, in run_publish
        self._hook_instance.publish(task_settings, item)
      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-publish2/v2.0.x/hooks/basic/create_version.py", line 241, in publish
        colorspace
      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-reviewsubmission/v0.3.x/app.py", line 147, in render_and_submit_path
        color_space)
      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-reviewsubmission/v0.3.x/python/tk_multi_reviewsubmission/renderer.py", line 146, in render_movie_in_nuke
        last_frame, version, name, color_space)
      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-reviewsubmission/v0.3.x/python/tk_multi_reviewsubmission/renderer.py", line 84, in gather_nuke_render_info
        nuke_version_major=nuke_version_major)
      File "/dd/tools/cent6_64/package/sgtk_core/2.13.1/python/tank/platform/bundle.py", line 712, in execute_hook_method
        **kwargs
      File "/dd/tools/cent6_64/package/sgtk_core/2.13.1/python/tank/hook.py", line 605, in execute_hook_method
        hook = create_hook_instance(hook_paths, parent, base_class=base_class)
      File "/dd/tools/cent6_64/package/sgtk_core/2.13.1/python/tank/hook.py", line 696, in create_hook_instance
        alternate_base_classes=alternate_base_classes
      File "/dd/tools/cent6_64/package/sgtk_core/2.13.1/python/tank/util/loader.py", line 69, in load_plugin
        raise TankLoadPluginError(message)
    TankLoadPluginError: Failed to load plugin /dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-reviewsubmission/v0.3.x/hooks/codec_settings.py. The following error was reported:
    Exception: <type 'exceptions.ImportError'> - No module named nuke
    Traceback (most recent call last):
      File "/dd/tools/cent6_64/package/sgtk_core/2.13.1/python/tank/util/loader.py", line 55, in load_plugin
        module = imp.load_source(module_uid, plugin_file)

      File "/dd/shows/TRAINING/user/work.raparna/tools/cent6_64/package/sgtk_apps/2.7.3/tk-multi-reviewsubmission/v0.3.x/hooks/codec_settings.py", line 18, in <module>
        import nuke

     // 
```